SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/DataEigen.h

```diff
@@ -0,0 +1,323 @@
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#ifndef SOFA_DEFAULTTYPE_EIGENMATRIXXD_H
+#define SOFA_DEFAULTTYPE_EIGENMATRIXXD_H
+#include <iostream>
+#include <sofa/defaulttype/config.h>
+
+#include <sofa/defaulttype/Vec.h>
+#include <sofa/defaulttype/Quat.h>
+#include <sofa/defaulttype/Mat.h>
+#include <sofa/helper/fixed_array.h>
+#include <sofa/helper/vector.h>
+#include <iostream>
+#include <map>
+
+#include <Eigen/Core>
+
+/////Eigen::MatrixXd
+///
+std::ostream & operator<< (std::ostream &out, const Eigen::MatrixXd* df)
+{
+    out<<(*df);
+    return out;
+}
+
+
+std::ostream & operator<< (std::ostream &out, const Eigen::MatrixXd& df)
+{
+    out<<(df);
+    return out;
+}
+
+std::ostream & operator<< (std::ostream &out, const Eigen::VectorXd* df)
+{
+    out<<(*df);
+    return out;
+}
+
+std::ostream & operator<< (std::ostream &out, const Eigen::VectorXd& df)
+{
+    out<<(df);
+    return out;
+}
+
+
+namespace sofa
+{
+
+namespace core::objectmodel
+{
+/////Eigen::MatrixXd
+///
+std::istream & operator>> (std::istream &in, Eigen::MatrixXd* df)
+{
+    in.setstate(std::ios_base::failbit) ;
+    return in;
+}
+
+
+std::istream & operator>> (std::istream &in, Eigen::MatrixXd& df)
+{
+    in.setstate(std::ios_base::failbit) ;
+    return in;
+}
+
+///Eigen::VectorXd
+
+std::istream & operator>> (std::istream &in, Eigen::VectorXd* df)
+{
+    in.setstate(std::ios_base::failbit) ;
+    return in;
+}
+
+std::istream & operator>> (std::istream &in, Eigen::VectorXd& df)
+{
+    in.setstate(std::ios_base::failbit) ;
+    return in;
+}
+
+}
+
+namespace defaulttype
+{
+
+struct EigenMatrixxDTypeInfo
```

Comment:
Uniform type name with Eigen, i.e EigenMatrixXdTypeInfo instead of  EigenMatrixxDTypeInfo, as it is for Eigen::MatrixXd (same for VectorXd)